src/widget.ts

```diff
@@ -157,7 +183,15 @@ export class TileLayerView extends WidgetView {
 
   url_changed() {
     // TODO React on url change!!
-  }
+    const newUrl = this.model.get('url');
+    if (newUrl) {
+      const newSource = new XYZ({
+        url: newUrl
+      });
 
-  tileLayer: TileLayer<OSM>;
+      this.tileLayer.setSource(newSource);
```

Comment:
I'm not sure I understand what this line is doing here, it seems to be at the same level as the methods and attributes? You  may want to remove this line.